graph/src/data/store/scalar.rs

```diff
@@ -167,7 +319,7 @@ impl BigInt {
         }
         let mut byte_array = if exp >= 0.into() { [0; 8] } else { [255; 8] };
         byte_array[..bytes.len()].copy_from_slice(&bytes);
-        BigDecimal::new(self.0, -i64::from_le_bytes(byte_array))
```

Comment:
I reviewed all the callers of this. Some places enforce an exponent of 0 (resulting often in non-normalized values) but those places then use the decimal in a way that shouldn't be a problem for consistency. Other places that use this code immediately do something with the decimal so that the result is normalized (like divide). I wonder if it's a  🐛💣  to not normalize here though.

Taking that further, should normalization just move into `new`?. I tried to evaluate each code path and everything looks good now, but it's unintuitive that the normalization happens via round-tripping through the underlying type. This structure lead to more code needing to be checked than necessary.